queries.sql

```diff
@@ -16,58 +16,112 @@ SELECT * FROM animals WHERE name = 'Gabumon';
 COMMIT;
+--
 
 SELECT * FROM animals;
-
+-- [X] Delete all recordes
 BEGIN;
-
 DELETE FROM animals;
-
 ROLLBACK;
+--
 
-SELECT * FROM animals;
-
+-- Inside a transaction:
 BEGIN;
-
+-- [X] Delete all animals born after Jan 1st, 2022.
 DELETE FROM animals WHERE date_of_birth > '2022-01-01';
-
+-- [X] Create a savepoint for the transaction.
 SAVEPOINT multitransactions;
-
+-- [X] Update all animals' weight to be their weight multiplied by -1.
 UPDATE animals SET weight_kg = weight_kg * -1;
-
+-- [X] Rollback to the savepoint
 ROLLBACK TO multitransactions;
-
+-- [X] Update all animals' weights that are negative to be their weight multiplied by -1.
 UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
-
+-- [X] Commit transaction
 COMMIT;
+--
 
-SELECT * FROM animals;
-
+-- [X] How many animals are there?
 SELECT count(name) FROM animals;
 
+-- [X] How many animals have never tried to escape?
 SELECT * FROM animals WHERE escape_attempts = 0;
 
+-- [X] What is the average weight of animals?
 SELECT avg(weight_kg) FROM animals;
 
+-- [X] Who escapes the most, neutered or not neutered animals?
 SELECT neutered AS isNeutered FROM animals WHERE escape_attempts = (SELECT max(escape_attempts) FROM animals);
 
+-- [X] What is the minimum and maximum weight of each type of animal?
 SELECT species, min(weight_kg) AS minimumWeight, max(weight_kg) AS maximumWeight FROM animals GROUP BY species;
 
+-- [X] What is the average number of escape attempts per animal type of those born between 1990 and 2000?
 SELECT species, CAST(avg(escape_attempts) AS decimal(10)) AS averageEscape FROM animals
 WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
 GROUP BY species;
+
+-- Write queries (using JOIN) to answer the following questions:
+-- [X] What animals belong to Melody Pond?
+SELECT name AS AnimalName, full_name AS Owner
+FROM animals INNER JOIN owners ON animals.owners_id = owners.id
+WHERE owners.full_name = 'Melody Pond';
+
+-- [X] List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.name AS AnimalName, species.name AS Species
+FROM animals
+INNER JOIN species
+ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
+
+-- [X] List all owners and their animals, remember to include those that don't own any animal.
+SELECT owners.full_name AS Owner, animals.name AS AnimalName
+FROM owners
+LEFT JOIN animals
+ON owners.id = animals.owners_id;
+
+-- [X] How many animals are there per species?
+SELECT COUNT(animals.name) AS AnimalsCount, species.name AS Species
+FROM animals
+INNER JOIN species
+ON animals.species_id = species.id
+GROUP BY species.name;
+
+-- [X] List all Digimon owned by Jennifer Orwell.
+SELECT owners.full_name AS Owner, animals.name As Animal, species.name AS Species
+FROM owners
+INNER JOIN animals
+ON owners.id = animals.owners_id
+INNER JOIN species
+ON species.id = animals.species_id
+WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
```

Comment:
- [ ] To correctly write this query, you are supposed to select  from the animals table, then joining the owners and species table respectively. You could consider writing it as below

```
SELECT * FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   LEFT JOIN species
   ON animals.species_id = species.id
   WHERE owners.full_name = 'Jennifer Orwell'
   AND species.name = 'Digimon';
```

**Note:** you do not have to explicitly write is as mine is.